R/getPrevalence.R

```diff
@@ -580,19 +590,42 @@ setMethod("agglomerateByPrevalence", signature = c(x = "SummarizedExperiment"),
             if(!is.null(rank)){
                 rowData(other_x)[,rank] <- other_label
             }
-            # temporary fix until TSE supports rbind
-            class <- c("SingleCellExperiment","RangedSummarizedExperiment",
-                       "SummarizedExperiment")
-            class_x <- class(x)
-            if(!(class_x %in% class)){
-                class <- "SingleCellExperiment"
-            } else {
-                class <- class[class_x == class]
-            }
-            x <- rbind(as(x[f,],class),
-                       as(other_x,class))
-            x <- as(x,class_x)
+            x <- rbind(x[f,], other_x)
         }
         x
     }
 )
+
+#' @rdname agglomerate-methods
+#' @export
+setMethod("agglomerateByPrevalence", 
+          signature = c(x = "TreeSummarizedExperiment"),
+    function(x, rank = NULL, other_label = "Other", agglomerate.tree = FALSE, 
+            mergeRefSeq = FALSE, ...){
+      x <- callNextMethod()
+      # input check
+      if(!.is_a_bool(agglomerate.tree)){
+        stop("'mergeTree' must be TRUE or FALSE.", call. = FALSE)
+      }
+      if(!.is_a_bool(mergeRefSeq)){
+        stop("'mergeRefSeq' must be TRUE or FALSE.", call. = FALSE)
+      }
+      # for optionally merging referenceSeq
+      f <- rownames(x) 
+      refSeq <- NULL
+      if(mergeRefSeq){
+        refSeq <- referenceSeq(x)
+      }
+      #
+      x <- .merge_rows(x, f, archetype = 1L, ...)
```

Comment:
Wondering what is the .merge_rows doing here.

I think this line should be the callNextMethod() call

Reply:
This merges rows with rownames --> this does not do anything?

Reply:
Yes I did not notice that, I removed it but I already call callNextMethod() at the beginning of the method, should I move it at this line?

Reply:
I did not check in detail, but in theory reference sequence might already be modified when calling callNextMethod(). That is because the SE method then calls other agglomeration methods.

That is why it might be safest to first take the reference sequences and then call callNextMethod() so that you have stored the original reference sequences.

Moreover, I realized that these all agglomeration functions are in same documentation page. agglomerateByVariable() can be done also in column-wise which means that it must have mergeRefSeq parameter as "hidden" parameter. --> agglomerateByPrevalence() should also have the parameter as hidden (in "...")

Reply:
You could do it with `list(...)[["mergeRefSeq"]]` without creating an internal function. Just explain in code comment, why this is done that way